File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int max = 0;
        while(right>left){
            int area = right-left;
            if(height[left]<height[right]){
                area *= height[left];
                left++;
            }
            else{
                area *= height[right];
                right--;
            }
            if(area>max)
                max = area;
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int heightSize) {
    int left = 0;                // Left pointer starting from the leftmost edge
    int right = heightSize - 1;  // Right pointer starting from the rightmost edge
    int maxWater = 0;            // Initialize the maximum water capacity

    while (left < right) {
        // Calculate the width of the container
        int width = right - left;

        // Calculate the height of the container (the minimum height between the two lines)
        int h = height[left] < height[right] ? height[left] : height[right];

        // Calculate the water capacity of the current container
        int water = width * h;

        // Update the maximum water capacity if the current container holds more water
        if (water > maxWater) {
            maxWater = water;
        }

        // Move the pointers towards each other
        if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxWater;
}